packages/compiler/src/server/types.ts

```diff
@@ -70,6 +70,14 @@ export interface CompileResult {
   readonly optionsFromConfig: CompilerOptions;
 }
 
+export interface CustomCompileResult {
+  readonly hasError: boolean;
+  readonly warningDiagnostics?: string[];
```

Comment:
why not returning the actual diagnostics.

This is way too specific for your current use case right now.

A reminder that adding this is now a new API that despite not exposing the type is now accessible to everybody.

Reply:
updated to return diagnotics object